ykrt/src/compile/jitc_yk/jit_ir.rs

```diff
@@ -211,13 +211,14 @@ impl Module {
     }
 
     /// Push an instruction to the end of the [Module].
-    ///
-    /// # Panics
-    ///
-    /// If `instr` would overflow the index type.
-    pub(crate) fn push(&mut self, inst: Inst) {
-        assert!(InstIdx::new(self.insts.len()).is_ok());
```

Comment:
The `assert!` here makes `push()` crash if the index overflows, so the commit message isn't accurate, right? ("Although previously the code didn't actually panic")

I do actually prefer rippling the error up though, so the change is good, but the commit message isn't.

Reply:
Oops, yes, you're right about the `panic`!